chapters/functions.tex

```diff
@@ -1762,6 +1762,72 @@ \section{Function Inlining and Event Generation}\label{function-inlining-and-eve
 
 If the function is called in a context where events will not be generated (e.g., inside another function without {\lstinline!GenerateEvents = true!}) no special action is needed.
 \end{nonnormative}
+\begin{example}
+The annotation also impacts the variability, \cref{variability-of-expressions} consider a trivial case.
+\begin{lstlisting}[language=modelica]
+function greaterEqual
+  input Real x1;
+  input Real x2;
+  output Boolean b;
+algorithm
+  b := x1 >= x2;
+  annotation(GenerateEvents=true, LateInline=true);
+end greaterEqual;
+
+function greaterEqualWO
+  input Real x1;
+  input Real x2;
+  output Boolean b;
+algorithm
+  b := x1 >= x2;
+  annotation(LateInline=true);
+end greaterEqualWO;
+
+model M
+  Boolean b1 = greaterEqual(time, 1);
+  Boolean b2 = greaterEqualWO(time, 1) "Illegal";
+  Boolean b3 = noEvent(greaterEqual(time, 1)) "Illegal";
+  Boolean b4 = greaterEqualWO(if b1 then 2 else 0, 1);
+  Real z = if greaterEqualWO(time, 1) then 2 else 1;
+end M;
+
```

Comment:
For a more digestible example, couldn't we have the text explaining the above part of the listing here?
```suggestion
\end{lstlisting}

 \lstinline!M! there will be an event when \lstinline!time! becomes greater or equal to \lstinline!1!, and thus the discrete-time variable \lstinline!b1! can be bound to this function call even though it involves the non-discrete time variable \lstinline!time!.
 Both \lstinline!b2! and \lstinline!b3! are illegal, since the right hand sides have non-discrete-time variability.
 The variable \lstinline!z! is legal, since a Real variable can have non-discrete time variability.
 The variable \lstinline!b4! is legal, since the function inputs have discrete-time variability.
 
\begin{lstlisting}[language=modelica]
```